PredixMobileUIKit/PredixBarChartView.swift

```diff
@@ -0,0 +1,330 @@
+    public init(_ values: [Double], label: String, colors: [UIColor]) {
+        self.values = values
+        self.label = label
+        self.colors = colors
+    }
+
+    /// Initialize the values and label of the bar.
+    /// A default color gray is used if color is not provided.
+    /// - parameter Values: the values the bar hold.
+    /// - parameter label: the bar label.
+    public init(_ values: [Double], label: String) {
+        self.values = values
+        self.label = label
+        colors = [NSUIColor.gray]
+    }
+}
+
+// MARK: - The PredixBarChartView Class
+
+/// PredixBarChartView -- Bar Chart
+@IBDesignable
+open class PredixBarChartView: BarChartView {
+
+    // MARK: PredixBarCharView Internal Variables
+
+    internal var limitLine = ChartLimitLine()
+    var xAxisValues: [String] = []
+    var bars: [Bar] = []
+
+    // MARK: PredixBarChartView Initial Values
+
+    /// :nodoc:
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        initialize()
+    }
+
+    /// :nodoc:
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        initialize()
+    }
+
+    /// Predix Mobile Bar chart initial values
+    fileprivate func initialize() {
+        initLegend()
+        initXaxis()
+        initYaxis()
+        chartDescription?.enabled = false
+    }
+
+    /// Initial legend
+    internal func initLegend() {
+        let legend = self.legend
+        legend.enabled = true
+        legend.horizontalAlignment = .right
+        legend.verticalAlignment = .top
+        legend.orientation = .vertical
+        legend.drawInside = true
+        legend.yOffset = 10.0
+        legend.xOffset = 10.0
+        legend.yEntrySpace = 0.0
+    }
+
+    /// Initial x axis
+    internal func initXaxis() {
+        let xaxis = xAxis
+        xaxis.valueFormatter = IndexAxisValueFormatter()
+        xaxis.drawGridLinesEnabled = true
+        xaxis.labelPosition = .bottom
+        xaxis.granularity = 1
+    }
+
+    /// Initial y axis
+    internal func initYaxis() {
+        let yLeftAxis = leftAxis
+        yLeftAxis.spaceTop = 0.35
+        yLeftAxis.axisMinimum = 1
+        yLeftAxis.drawGridLinesEnabled = false
+
+        let yRightAxis = rightAxis
+        yRightAxis.spaceTop = 0.35
+        yRightAxis.axisMinimum = 1
+        yRightAxis.drawGridLinesEnabled = false
+    }
+
+    // MARK: PredixBarChartView Adding Displaying or Removing a Limit Line
+
+    /// Add a horizontal limit line on the bar chart. The line in the chart marks a certain maximum or limit on the x axis).
+    /// - parameter limit: number of where the line should be drawn horizontally.
+    /// - parameter label: name of the label line.
+    public func addALimit(limit: Double, label: String) {
+        limitLine = ChartLimitLine(limit: limit, label: label)
+        rightAxis.addLimitLine(limitLine)
+        setNeedsDisplay()
+    }
+
+    /// Display the horizontal limit line added into the chart.
```

Comment:
Confused about the functionality here. If calling displayLimitLine(false) removes the limit line, what happens If I then call displayLimitLine(true)? Can I toggle the limit line in this way? If so, update documentation comment to something about showing and hiding the limit line. Otherwise consider updating the method's functionality.